packages/transport/src/chrome-runtime/adapter.ts

```diff
@@ -0,0 +1,86 @@
+import { Any, AnyMessage, JsonValue, MethodKind, ServiceType } from '@bufbuild/protobuf';
+import {
+  CallOptions,
+  ConnectError,
+  Code as ConnectErrorCode,
+  ConnectRouter,
+  ContextValues,
+  PromiseClient,
+  ServiceImpl,
+  createPromiseClient,
+  createRouterTransport,
+} from '@connectrpc/connect';
+import { typeRegistry } from '@penumbra-zone/types/src/registry';
+import { iterableToStream, MessageToJson } from '../stream';
+
+interface ChromeRuntimeCallOptions {
+  timeoutMs?: number;
+  signal?: AbortSignal;
+  contextValues?: ContextValues;
+}
+
+/**
+ * This method creates a simple entry function into any service. The entry
+ * accepts a serialized json request, annotated with any type handled by the
+ * service, and returns a serialized single response or stream of responses.
+ *
+ * @param serviceType ServiceType of the service
+ * @param impl Object containing an implementation of the service
+ * @param remote Client to a remote service, for fallback/proxy
+ * @returns (req: JsonValue) => Promise<JsonValue> | ReadableStream<JsonValue>
+ */
+export const adaptServiceImpl = <S extends ServiceType>(
+  serviceType: S,
+  impl: Partial<ServiceImpl<S>>,
+  //ctx?: ContextValues,
```

Comment:
this could provide a "global context" i cut it out because i was uncertain about wide reuse of a context object. likely also a problem at service-worker.ts

 should probably use ContextInit type